spec/requests/oauth_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+
+RSpec.describe Oauth::OauthController, type: :request do
+  let(:valid_client) {
+    OauthClient.create(
+          client_id: '123',
+          client_name: 'Rubber Toes',
+          redirect_uri: 'https://www.robert.com/callback'
+        )
+      }
+
+  def make_request(params)
+    get oauth_authorize_path, params: params
+  end
+
+  def expect_bad_request(response)
+    expect(response).to have_http_status(:bad_request)
+  end
+
+  describe 'GET /oauth/authorize' do
+    context 'when request is valid' do
```

Comment:
I would also put your arguments into keyword variables that get passed implicitly just as a general practice, as well as setting defaults so that you're only declaring arguments that matter to the test, it makes the code easier to read.

Reply:
okay makes sense, I just pushed this up.